### PR TITLE
refactor: template pages

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Refactors the structure of pages use by templates into: /library (for public) and /templates (for publishers)

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/28980303/todos/5252812103)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Can visit pages
> - /library
> - /library/[journeyId]
> - /templates
> - /templates/[journeyId]
> - /templates/[journeyId]/edit

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/journeys-admin/pages/templates/[journeyId]/edit.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { PageWrapper } from '../../../src/components/PageWrapper'
+import i18nConfig from '../../../next-i18next.config'
+
+function TemplateEditPage(): ReactElement {
+  const AuthUser = useAuthUser()
+
+  return (
+    <>
+      <NextSeo title="Template Details" />
+      <PageWrapper title="Template Details" authUser={AuthUser}>
```
Comment (storyworks) [inline apps/journeys-admin/pages/templates/[journeyId]/edit.tsx:20]: Update title

File: apps/journeys-admin/pages/templates/[journeyId].tsx [hunk 2]
```diff
@@ -11,13 +11,13 @@ import { getLaunchDarklyClient } from '@core/shared/ui/getLaunchDarklyClient'
 import { PageWrapper } from '../../src/components/PageWrapper'
 import i18nConfig from '../../next-i18next.config'
 
-function TemplateDetails(): ReactElement {
+function TemplateDetailsAdmin(): ReactElement {
   const AuthUser = useAuthUser()
   return (
     <>
       <NextSeo title="Template Details" />
       <PageWrapper title="Template Details" authUser={AuthUser}>
```
Comment (storyworks) [inline apps/journeys-admin/pages/templates/[journeyId].tsx:19]: Double check this, do we even have a design for it?

File: apps/journeys-admin/pages/library/[journeyId].tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { getLaunchDarklyClient } from '@core/shared/ui/getLaunchDarklyClient'
+import { PageWrapper } from '../../src/components/PageWrapper'
+import i18nConfig from '../../next-i18next.config'
+
+function TemplateDetails(): ReactElement {
+  const AuthUser = useAuthUser()
+  return (
+    <>
+      <NextSeo title="Template Details" />
+      <PageWrapper title="Template Details" authUser={AuthUser}>
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/pages/library/[journeyId].tsx:19]: Changed to "Journey Template"

File: apps/journeys-admin/pages/templates/[journeyId]/edit.tsx (hunk 1 above)
Comment (csiyang) [reply apps/journeys-admin/pages/templates/[journeyId]/edit.tsx:20]: Changed this to "Edit Template"

File: apps/journeys-admin/pages/templates/[journeyId].tsx (hunk 2 above)
Comment (csiyang) [reply apps/journeys-admin/pages/templates/[journeyId].tsx:19]: Still trying to resolve what name this page needs to be, but can be updated when the components for this page gets added in.

**Review by storyworks — APPROVED** [review]

**Pull request merged**